django/contrib/admin/static/admin/js/SelectFilter2.js

```diff
@@ -155,9 +155,25 @@ Requires core.js and SelectBox.js.
 
             if (!is_stacked) {
                 // In horizontal mode, give the same height to the two boxes.
+                let height;
                 const j_from_box = document.getElementById(field_id + '_from');
                 const j_to_box = document.getElementById(field_id + '_to');
-                let height = filter_p.offsetHeight + j_from_box.offsetHeight;
+                const collapsed_fieldsets = Array.from(document.querySelectorAll('fieldset.collapsed'));
+
+                const is_selectbox_collapsed = collapsed_fieldsets.some(function(collapsed_fieldset) {
```

Comment:
isn't there too much logic in this one function?
is_selectbox_collpased:
 1) checking that the selectboxes are collapsed and returns the state of it
 2) switching collapsed state
 3) calculating the height

Reply:
I have the same doubts. I think a separate function is unnecessary.

Reply:
Should I replace the `.some` method with a `for...of` loop? It will look like this. 
```
let height = filter_p.offsetHeight + j_from_box.offsetHeight;

const collapsed_fieldsets = Array.from(document.querySelectorAll('fieldset.collapsed'));
for (const collapsed_fieldset of collapsed_fieldsets) {
        // Collapsed fieldset contains the selectbox
        if (collapsed_fieldset.querySelector('.field-' + field_name) !== null) {
                collapsed_fieldset.classList.remove('collapsed');
                height = filter_p.offsetHeight + j_from_box.offsetHeight;
                collapsed_fieldset.classList.add('collapsed');
                break;
       }
}
```

Reply:
Also, do we need to iterate? We could use `querySelector()`, e.g.
```js
let height;
const collapsed_field = document.querySelector('fieldset.collapsed .field-' + field_name);
if (collapsed_field !== null) {
    collapsed_field.parentNode.classList.remove('collapsed');
    height = filter_p.offsetHeight + j_from_box.offsetHeight;
    collapsed_field.parentNode.classList.add('collapsed');
} else {
    height = filter_p.offsetHeight + j_from_box.offsetHeight;
}
```

Reply:
There could be more then one fieldset with `collapse` class. In that case we will have to find which one contains the selectbox.

Reply:
> There could be more than one fieldset with `collapse` class. In that case we will have to find which one contains the selectbox.

But I added `field_name` to the selector.

Reply:
Oh yeah, you are right. This looks more simpler. I will make the change.